sdk/service-map/arm-servicemap/src/models/mappers.ts

```diff
@@ -211,7 +209,7 @@ export const CoreResource: msRest.CompositeMapper = {
       serializedName: "kind",
       clientName: "kind"
     },
-    uberParent: "CoreResource",
+    uberParent: "BaseResource",
```

Comment:
@onalant - with this change, it should work. Thus resolving this comment https://github.com/Azure/autorest.typescript/issues/458#issuecomment-515614561

Reply:
I think it was fine originally. I was more commenting on the fact that `CoreResource` was referred to as `BaseResource.CoreResource` in the discriminators, whereas other ``base classes'' are usually given their own name.

The discriminators for `CoreResource` subtypes are still not as expected:
```
  'BaseResource.machine' : Machine,
...
  'BaseResource.process' : Process,
  'BaseResource.port' : Port,
  'BaseResource.clientGroup' : ClientGroup,
  'BaseResource.machineGroup' : MachineGroup,
```

Reply:
Just verified by writing a sample and everything works as expected.
In fact I modified this operation signature `machineGroups.create` to take machineGrooup of type `CoreResourceUnion` (Base class) and updated the `createOperationSpec` mapper to take the requestBody of type `CoreResource` mapper
```ts
  create(resourceGroupName: string, workspaceName: string, machineGroup: Models.CoreResourceUnion, options?: msRest.RequestOptionsBase | msRest.ServiceCallback<Models.MachineGroup>, callback?: msRest.ServiceCallback<Models.MachineGroup>): Promise<Models.MachineGroupsCreateResponse> {
```
And I provided MachineGroup as the request body and the sdk was able to correctly find out the mapper for MachineGroup from the discriminators map by looking for `BaseResource.machineGroup`.

sample
```ts
import * as msRestNodeAuth from "@azure/ms-rest-nodeauth";
import { ServicemapManagementClient, ServicemapManagementModels } from "@azure/arm-servicemap";
const subscriptionId = process.env["AZURE_SUBSCRIPTION_ID"] || "";

msRestNodeAuth.AzureCliCredentials.create().then((creds) => {
  const client = new ServicemapManagementClient(creds, subscriptionId);
  const resourceGroupName = "testresourceGroupName";
  const workspaceName = "testworkspaceName";
  const machineGroup: ServicemapManagementModels.MachineGroup = {
    displayName: "dp1",
    kind: "machineGroup"
  };
  client.machineGroups.create(resourceGroupName, workspaceName, machineGroup).then((res) => {
    console.log(res);
  });
}).catch((err) => {
  console.error(err);
});
```

I guess what you are missing is that in the swagger spec, ["CoreResource" is an allOf "Resource"](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/service-map/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/arm-service-map.json#L1769) and ["Resource" is tagged with extension `"x-ms-azure-resource": true`](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/service-map/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/arm-service-map.json#L1625). It is a microsoft swagger extension that Autorest understands. When it sees this extension it will make the model as a child of `BaseResource`. Definition of `BaseResource` is present in `@azure/ms-rest-azure-js` package (runtime).

Reply:
Ah, I missed the `allOf` declaration. Thank you for the explanation and your time w.r.t. this issue.

Reply:
Thanks for filing the original issue and taking the time for a deep dive in the generator and the generated code.